[Bug]: .NET 9.0 Alpha build fails in Windows 1809 Docker containers with Terminal Logger error

### Issue Description

We're seeing this issue consistently with the latest .NET 9.0 alpha builds in our Windows Server Core 2019 and Nanoserver 1809 SDK images. The project is created with the `dotnet new web` command, and started failing with the SDK version 9.0.100-alpha.1.24058.1. The last known good SDK that we have is 9.0.100-alpha.1.24053.3.

```
 MSBUILD : error MSB4015: The build stopped unexpectedly because the "TerminalLog
 ger" logger failed unexpectedly during shutdown.\x1b[32X\x1b[32C
 System.IO.IOException: The parameter is incorrect.\x1b[30X\x1b[30C
    at System.ConsolePal.SetConsoleOutputEncoding(Encoding enc)\x1b[18X\x1b[18C
    at System.Console.set_OutputEncoding(Encoding value)\x1b[25X\x1b[25C
    at Microsoft.Build.Logging.TerminalLogger.TerminalLogger.Shutdown()\x1b[10X\x1b[10C
    at Microsoft.Build.BackEnd.Logging.LoggingService.ShutdownLogger(ILogger logger)
```

Please excuse the weird text formatting - I didn't want to cut anything off in case it's important.

(here's a link to failing pipeline run that was pulled from: https://dev.azure.com/dnceng-public/public/_build/results?buildId=531306&view=logs&j=993563f7-f5c3-53cf-2b69-c3a3c7bb299e&t=4e281778-b663-5be4-ed90-ae0bb7f6fb0d&l=336)

### Steps to Reproduce

I am unable to reproduce this issue on my local machine, but we are experiencing it consistently in our CI. If there are any additional diagnostic steps you want me to perform, let me know.

### Expected Behavior

The build completes successfully.

### Actual Behavior

```
 MSBUILD : error MSB4015: The build stopped unexpectedly because the "TerminalLog
 ger" logger failed unexpectedly during shutdown.\x1b[32X\x1b[32C
 System.IO.IOException: The parameter is incorrect.\x1b[30X\x1b[30C
    at System.ConsolePal.SetConsoleOutputEncoding(Encoding enc)\x1b[18X\x1b[18C
    at System.Console.set_OutputEncoding(Encoding value)\x1b[25X\x1b[25C
    at Microsoft.Build.Logging.TerminalLogger.TerminalLogger.Shutdown()\x1b[10X\x1b[10C
    at Microsoft.Build.BackEnd.Logging.LoggingService.ShutdownLogger(ILogger logger)
```

### Analysis

_No response_

### Versions & Configurations

_No response_

--- a/src/MSBuild/TerminalLogger/Terminal.cs
+++ b/src/MSBuild/TerminalLogger/Terminal.cs
@@ -200,6 +200,15 @@ internal sealed class Terminal : ITerminal
     /// <inheritdoc/>
     public void Dispose()
     {
-        Console.OutputEncoding = _originalOutputEncoding;
+        try
+        {
+            Console.OutputEncoding = _originalOutputEncoding;
+        }
+        catch
+        {
+            // In some terminal emulators setting back the previous console output encoding fails.
+            // See https://github.com/dotnet/msbuild/issues/9662.
+            // We do not want to throw an exception if it happens, since it is a non-essentual failure in the logger.
+        }
     }
 }

